nilearn/_utils/cache_mixin.py

```diff
@@ -30,6 +30,62 @@
 __CACHE_CHECKED = dict()
 
 
+def _check_memory(memory, verbose=0):
+    """Function to make ensure a memory object.
+
+    Parameters
+    ----------
+    memory: None or instance of joblib.Memory or string
```

Comment:
May prefer to denote "str" rather than string since you denote integer as int below. For the sake of consistency.